### ISSUE TITLE
extra_script is missing in SLURM(BaseScheduler)

### ISSUE THREAD

**Issue by aeantipov** [basnijholt/adaptive-scheduler#58]
```
scheduler_kwargs = dict(
        num_threads = 4,
        extra_scheduler=[
            "--exclusive", 
            "--partition=partition"
        ],
        extra_script="umask 0000", # or ["umask 0000"]
        executor_type="process-pool",
    )
``` 
ends up having
```
#!/bin/bash
#SBATCH --ntasks 4
#SBATCH --no-requeue
#SBATCH --exclusive
#SBATCH --partition=partition

export MKL_NUM_THREADS=4
export OPENBLAS_NUM_THREADS=4
export OMP_NUM_THREADS=4
export NUMEXPR_NUM_THREADS=4

python ... 
```

Note the missing lines from extra_script

**Comment by basnijholt** [comment]
@jbweston implemented this 7 months ago. I have never used it myself. From looking at the source-code I have no clue as to why this is happening for you.

**Comment by basnijholt** [comment]
@aeantipov, I tried this:
```
scheduler_kwargs = dict(
        num_threads = 4,
        extra_scheduler=[
            "--exclusive", 
            "--partition=partition"
        ],
        extra_script="umask 0000", # or ["umask 0000"]
        executor_type="process-pool",
    )
import adaptive_scheduler
sch = adaptive_scheduler.scheduler.SLURM(cores=1, **scheduler_kwargs)
print(sch.job_script())
```
which prints:
```
#!/bin/bash
#SBATCH --ntasks 1
#SBATCH --no-requeue
#SBATCH --exclusive
#SBATCH --partition=partition

export MKL_NUM_THREADS=4
export OPENBLAS_NUM_THREADS=4
export OMP_NUM_THREADS=4
export NUMEXPR_NUM_THREADS=4


umask 0000

/Users/basnijholt/miniconda3/bin/python run_learner.py --n 1 --log-fname ${NAME}-${SLURM_JOB_ID}.log --job-id ${SLURM_JOB_ID} --name ${NAME}
```

So I think you might be using an old version.

**Comment by aeantipov** [comment]
It seems, it's a further dependency that's causing the issue.

**Comment by basnijholt** [comment]
I have noticed that this actually still doesn't work.

```
import concurrent.futures
ex = concurrent.futures.ProcessPoolExecutor()
# Ignores extra_script
run_manager.ioloop.run_in_executor(ex, scheduler.write_job_script, "test")
# Correctly uses extra_script
scheduler.write_job_script("test2")
```

It probably has to do with the serialization of the scheduler.

**Comment by basnijholt** [comment]
Right:
```
import pickle
print(scheduler.extra_script)  # prints correctly
print(pickle.loads(pickle.dumps(scheduler)).extra_script)  # prints empty string
```

**Comment by basnijholt** [comment]
Fixed by https://github.com/basnijholt/adaptive-scheduler/commit/e7eb7df63a7d48d14e900e161044503fc9ac97bd and I released v0.9.16.
